packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2675,47 +2681,47 @@ class WidgetInspector extends StatefulWidget {
 class _WidgetInspectorState extends State<WidgetInspector>
     with WidgetsBindingObserver {
 
-  _WidgetInspectorState() : selection = WidgetInspectorService.instance.selection;
+  _WidgetInspectorState();
 
   Offset? _lastPointerLocation;
 
-  final InspectorSelection selection;
+  InspectorSelection get selection =>
+      WidgetInspectorService.instance.selection;
 
-  /// Whether the inspector is in select mode.
-  ///
-  /// In select mode, pointer interactions trigger widget selection instead of
-  /// normal interactions. Otherwise the previously selected widget is
-  /// highlighted but the application can be interacted with normally.
-  bool isSelectMode = true;
+  bool get isSelectMode =>
+      WidgetInspectorService.instance.isSelectMode.value;
+  set isSelectMode(bool value) =>
```

Comment:
this is a bit of an overly picky style issue and perhaps just my preference.
I would avoid adding getters and setters that obfuscate what is going on. Code written like this makes it look like you have a stateful widget where you are not calling setState while modifying the state. Instead return just a getter that returns WidgetInspectorService.instance.isSelectMode
or inline the full lines.